### ISSUE TITLE
Connection timeout for Mi Robot Vacuum Cleaner cards

### ISSUE THREAD

**Issue by jghaanstra** [jghaanstra/com.xiaomi-miio#7]
Since Mi Robot firmware 3.3.9_003077 the Homey app seems to have trouble sending commands to the Mi Robot which according to some users time out.

Current implementation is live requests for all capabilities, possible solution could be polling the device and keeping track of its state. This would however increase the network traffic specially for a device that is only used once in a while.

Alternative could be to lengthen the request time out. This however effect the complete miio library.

Needs further research.

**Comment by chrisboer** [comment]
I got my Mi Robot last week and installed the latest firmware after this I have added it to homey so got the token but I can say I don't have these issues you have.

**Comment by jghaanstra** [comment]
@chrisboer thank you for reporting, I do not have these issues myself but two other users on this latest firmware have reported these timeouts. I have yet to investigate this further once I actually receive this latest firmware myself. But seeying your reaction it might be caused by something else.

**Comment by synegic** [comment]
I do have these issues, on the latest firmware. The status of the robot does not update in Homey and when changing operation mode in Flows, it does not work. Sometimes, however, when you change the device state directly it eventually works (after 10 seconds or so). But after it finishes, it never updates device state (always stays on 'Cleaning').

**Comment by jghaanstra** [comment]
I have confirmed this issue. It seems the Mi Robot needs longer to respond since firmware 3.3.9_003077 and requests often time out. I have submitted a request (https://github.com/aholstenson/miio/issues/59) to increase the time out in the miio library which is used in my app to control the Mi Robot. This will make it slow to respond but should at least work again. I'm not sure when this will become available though.
